routes/attribution.js

```diff
@@ -1,38 +1,56 @@
 const Joi = require('joi');
 
+const { knownLanguages, knownTypesOfUse, Attribution } = require('../models/attribution');
+const errors = require('../services/util/errors');
 const definitions = require('./__swagger__/definitions');
 const prefix = require('./__utils__/path')('/attribution');
 
 const routes = [];
 
-const attributionMock = {
-  license: 'CC BY-SA 3.0',
-  attribution_plain:
-    'Pierre Dalous (https://commons.wikimedia.org/wiki/File:Pair_of_Merops_apiaster_feeding.jpg), "Pair of Merops apiaster feeding", https://creativecommons.org/licenses/by-sa/3.0/legalcode',
-  attribution_text:
-    'Pierre Dalous (https://commons.wikimedia.org/wiki/File:Pair_of_Merops_apiaster_feeding.jpg), "Pair of Merops apiaster feeding", https://creativecommons.org/licenses/by-sa/3.0/legalcode',
-  attribution_html:
-    'Pierre Dalous (https://commons.wikimedia.org/wiki/File:Pair_of_Merops_apiaster_feeding.jpg), "Pair of Merops apiaster feeding", https://creativecommons.org/licenses/by-sa/3.0/legalcode',
-  license_url: 'https://creativecommons.org/licenses/by-sa/3.0/legalcode',
-};
+const attributionSchema = Joi.object()
+  .required()
+  .keys({
+    attributionHtml: Joi.string().required(),
+    attributionPlain: Joi.string().required(),
+    licenseId: Joi.string().required(),
+    licenseUrl: Joi.string()
+      .uri()
+      .required(),
+  })
+  .meta({ className: 'AttributionShowResponse' });
+
+function handleError(h, { message }) {
+  switch (message) {
+    case errors.invalidUrl:
+    case errors.validationError:
+      return h.error(message, { statusCode: 422 });
+    case errors.apiUnavailabe:
+      return h.error(message, { statusCode: 503 });
+    default:
+      return h.error(message);
+  }
+}
+
 routes.push({
-  path: prefix('/{language}/{file}'),
+  path: prefix('/{languageCode}/{file}/{typeOfUse}/unmodified'),
   method: 'GET',
   options: {
     description: 'Generate attribution',
     notes: 'Generate attribution hints for the given file.',
     validate: {
       params: {
-        language: Joi.string(),
+        languageCode: Joi.string().valid(knownLanguages),
         file: Joi.string(),
+        typeOfUse: Joi.string().valid(knownTypesOfUse),
       },
     },
     response: {
-      schema: Joi.object()
-        .required()
-        .meta({ className: 'AttributionShowResponse' }),
+      schema: attributionSchema,
       status: {
         400: definitions.errors['400'],
```

Comment:
Idea: put this globally somewhere 🤔

Reply:
we talked about this in person and decided to have a look at all routes and extract common behavior in a follow up